benches/main.rs

```diff
@@ -0,0 +1,272 @@
+#![cfg(feature = "benches")]
+#![feature(test)]
+
+extern crate test;
+
+use anyhow::{bail, Context as _, Result};
+use std::path::Path;
+use wasmtime::{Config, Engine, OptLevel, Store, Strategy};
+use wasmtime_wast::WastContext;
+use wast::{
+    parser::{self, ParseBuffer},
+    Wat,
+};
+
+macro_rules! mk_test {
+    ($(#[$attrs:meta])* $name:ident, $path:expr, $strategy:expr) => {
+        mod $name {
+            use wasmtime::Strategy;
+
+            #[bench]
+            $(#[$attrs])*
+            fn compile(b: &mut ::test::bench::Bencher) {
+                crate::bench_compile(b, $path, $strategy).unwrap();
+            }
+
+            #[bench]
+            $(#[$attrs])*
+            fn run(b: &mut ::test::bench::Bencher) {
+                crate::bench_run(b, $path, $strategy).unwrap();
+            }
+        }
+    };
+}
+
+include!(concat!(env!("OUT_DIR"), "/wast_testsuite_tests.rs"));
+
+fn bench_compile(b: &mut test::bench::Bencher, wast: &str, strategy: Strategy) -> Result<()> {
+    let path = Path::new(wast);
+
+    let simd = path.iter().any(|s| s == "simd");
+
+    let bulk_mem = path.iter().any(|s| s == "bulk-memory-operations");
+
+    // Some simd tests assume support for multiple tables, which are introduced
+    // by reference types.
+    let reftypes = simd || path.iter().any(|s| s == "reference-types");
+
+    let multi_val = path.iter().any(|s| s == "multi-value");
+
+    let mut cfg = Config::new();
+    cfg.wasm_simd(simd)
+        .wasm_bulk_memory(bulk_mem)
+        .wasm_reference_types(reftypes)
+        .wasm_multi_value(multi_val)
+        .strategy(strategy)?
+        .cranelift_debug_verifier(cfg!(debug_assertions));
```

Comment:
This I think is duplicated with the testing code as well as with the function below, could this all be consolidated into one function, perhaps in the `wasmtime-wast` crate itself?